### PR TITLE
UI to flag an add-on review for spam, etc

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/2774

Here is the main functionality:

addons-frontend-flag-review mov

screenshot 2017-10-20 08 54 06

You have to be logged in first or else you'll see this. This is a functional button that will log you in:
screenshot 2017-10-19 16 18 28

You can't flag your own review (duh):
screenshot 2017-10-19 16 39 12

If there's an error (like a 500 API error) you'll see the message and you can try again. Encountering an error is not something we expect to happen. I had to break the login code to show this message.
screenshot 2017-10-19 16 32 35

### CODE REVIEW

**Comment by kumar303** [comment]
r? @tofumatt or @willdurand 

Sorry for the epic patch.

File: tests/unit/amo/components/TestFlagReview.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      reviewId: review.id,
+    }));
+
+    // The user triggers an error:
+    errorHandler.handle(new Error('Unexpected API error'));
+
+    const root = render({ errorHandler, reason });
+    expect(root.find(ErrorList)).toHaveLength(1);
+
+    // It should still display a button so they can try again.
```
Comment (tofumatt) [inline tests/unit/amo/components/TestFlagReview.js]: 👍

File: tests/unit/amo/components/TestFlagReviewMenu.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+      const flag = menu.find('.FlagReviewMenu-flag-bug-support-item')
+        .find(FlagReview);
+      expect(flag).toHaveProp('review', review);
+      expect(flag).toHaveProp('reason', REVIEW_FLAG_REASON_BUG_SUPPORT);
+      expect(flag).toHaveProp('wasFlaggedText');
+      expect(flag).toHaveProp('buttonText');
+    });
+
+    it('does not prompt you to flag a response as a bug/support', () => {
```
Comment (tofumatt) [inline tests/unit/amo/components/TestFlagReviewMenu.js]: I get why this is but it just took me a few seconds to parse, understand the intention here, and then realise why. Feel free to disagree but I think a comment explaining why "responses can't be flagged as bug reports" might be handy. 😄

Comment (kumar303) [reply tests/unit/amo/components/TestFlagReviewMenu.js]: ok, I added a comment to `FlagReviewMenu` about it.

File: src/ui/components/TooltipMenu/styles.scss [hunk 3]
```diff
@@ -48,31 +48,42 @@
...
-      @include font-medium();
-
+    a:link,
+    button {
       color: $grey-90;
-      cursor: pointer;
       font-size: $font-size-m-smaller;
       line-height: 1.6;
+      overflow: hidden;
+      text-align: left;
```
[the remark this answers is not in the document]
Comment (kumar303) [reply src/ui/components/TooltipMenu/styles.scss]: whoops, that was a mistake

**Comment by kumar303** [comment]
Thanks for the review!

> are we allowed to flag a review as multiple things?

The API does allow it and I guess you might want to flag a review for both spam and inappropriate language -- maybe flagging it twice would raise its priority for admins? Or maybe flagging it once would be enough.

Anyway, I was flying blind without UX specs here so we will definitely need to iterate on this. I'm also not sure if the menu should auto-close when you're done flagging + lots of little things.

**Pull request merged**